### DISCUSSION TITLE
How to use import statement?

### DISCUSSION THREAD

**Discussion by ngekoding** [vuepress/core#219] [Q&A] [answered]
I just try the basic, and want to use import to make the project structure better.

Here is the project stucture.
```
.
├── .vuepress
│   ├── config.js
│   └── configs
│       ├── index.js
│       └── sidebar
│           ├── id.js
│           └── index.js
├── README.md
└── guide
    └── README.md
```

I want to import the sidebar from config.js, and get the error below:

```js
// docs/.vuepress/config.js
import { sidebar } from './configs'
```

<img width="598" alt="VuePress import error" src="https://user-images.githubusercontent.com/11625690/121874747-3e2f0c00-cd3a-11eb-89b9-13d9b1d50177.png">

And here is my project https://github.com/ngekoding/vuepress-starter

Am I doing something wrong? 

Thanks in advance

**Comment by yzx9** [comment, marked as answer]
This is a limiation from `node.js`, mix `import` and `require` will get an error. The simplest solution is that change `.js` to `.ts`, typescript will help you solve the the problem.

**Reply by ngekoding** [reply]
Thanks for the answer.
So there is no way to use import with .js? 
I'm afraid that will take a lot of time to understanding TypeScript or that just my thinking?

**Reply by yzx9** [reply]
> So there is no way to use import with .js?

As I remember, we are using `require` in node side, so you have to make a workaround. There are some feasible solutions, like babel, but it also bring complexity, **use `require` instead of `import/export`** did not introduce anything new and solve your issue🤔

>  I'm afraid that will take a lot of time to understanding TypeScript or that just my thinking?

Haha😄, actually not, we have built-in `esbuild` which will help you transform `.ts` to `.js`, and typescript is a superset of javascript, all js code is legal ts code, like c in c++

**Reply by ngekoding** [reply]
@nsznsznjsz Thank you, I change it to use `require` and the error was gone. 
But, I don't know how to replace the export part with require.

How to change this part?

```js
export * as sidebar from './sidebar'
```

Sorry if this is a basic, but I can't found it yet on Google.

**Reply by yzx9** [reply]
@ngekoding Feel free to ask any question🙌, try following code:

```js
// From:
export * as sidebar from './sidebar'
// To:
const sidebar = require('./sidebar')
exports.sidebar = sidebar
```

**Reply by ngekoding** [reply]
Thank you, the problem solved!
